### DISCUSSION TITLE
Do you plan on making this OS usable? Or just a hobby?

### DISCUSSION THREAD

**Discussion by xplshn** [vinc/moros#492] [General]

**Comment by vinc** [comment]
Hi @xplshn, can you define usable for you? Would you mind listing what's missing for that? I'm really interested about getting feedback on what I'm building.

For me the next important milestone for the kernel is finishing the work to allow userspace programs to allocate memory. But I don't want the OS to be just a small kernel that can run the usual Unix tools.

I want to create a polished terminal environment without the constrain of being a direct clone of Unix or anything else. I grew up coding in BASIC on an IBM PC running DOS, and I'm fascinated by the history of Unix, ITS, and Lisp. I'm also inspired by the idea of permacomputing so I'm targeting computers from around 2010 that are now pretty cheap and still powerful enough to do everything we might want from a terminal. That's why I'm also spending time coding a lisp-like language, a text editor, network tools, and many other things inside the OS.

But MOROS is just a hobby, it won’t be big and professional like Linux. There's plenty of other bigger and better alternative operating systems already! One important goal is to inspire other people to start their own hobby OS to increase the diversity of the ecosystem.

**Comment by xplshn** [comment]
I thought you had the objective of making MOROS a minimalist, computing OS. For simple tasks.

If this were to be a usable in the day to day OS, it'd hove at least a WM like Sway, or something more complicated to port like Xorg, and a minimalistic package manager to build recipes or download binaries. An important addition would be a web browser, like netsurf or neosurf, a TUI mail client, a terminal emulator like foot. An image viewer like swayimg, all little utilities.

I think that the important bit of a project, is that the creator likes it. After all, its all yours.

**Reply by vinc** [reply]
Thanks for the feedback!

I'd suggest looking at Redox or Serenity that are big enough to provide that kind of rich environment! MOROS is much smaller, multiple order of magnitudes smaller, and it has no GUI with no plan to ever have one, you can think of it has a modern DOS instead of Windows.

My plan is nonetheless to make it usable for simple tasks. I use it everyday on a few old computers to write text with `edit`, check the weather from my station with `http`, and do some computations with `calc` or `lisp`. I have plans to simplify browsing the web or sending emails with small text utilities, but I have not tried yet to read an IMAP server for example to have a more advanced mail client. I'm also waiting for a rust implementation of TLS that will work without a standard environment to get access to HTTPS and Gemini without a proxy.

I'll use this conversation and a few others to make an about page on the website to better describe the project now that I have a clearer vision for what I want to achieve with it :)

**Reply by xplshn** [reply]
I'd love to see something like Peropesis, a personal computing OS that is 100% TUI/CLi based. But its sad they went the SystemD route, use GlibC, and aren't using smaller/simpler and more Unix compliant libraries and tools.

**Reply by vinc** [reply]
Have you looked into Alpine Linux? It doesn't use SystemD and GlibC. I've been thinking about adding a `Dockerfile` to describe the bedrock platform for Rust to build MOROS and Alpine is the most simple Linux distribution for that.

https://www.alpinelinux.org/
https://permacomputing.net/bedrock_platform/

**Comment by xplshn** [comment]
Nonetheless, this is an awesome project and I can't believe one person buil t it all in Rust

**Reply by vinc** [reply]
Thanks! I like the level of complexity of what I had to do over the past 2.5 years so far, it's not that hard, but at the same time it's just the right amount of challenge for a hobby with a few hours of free time per week. You can spend one month on the disk driver, then another on a network card driver, then work on the editor or the lisp for a change. It's perfect haha.
